src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/utils/user_config_validatator.py

```diff
@@ -74,6 +101,60 @@ def block_DHCP_config(self):
         if "isStatic" in self.__config and not self.__config["isStatic"]:
             raise InvalidInputError("isStatic should be true. Only Static IP configuration is supported currently")
 
+    def check_if_old_nw_config(self):
+        if not self.__config["applianceControlPlaneIpAddress"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["networkForApplianceVM"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolStart"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolEnd"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["gatewayIPAddress"].strip():
+            return False
+        return True
+
+    def check_if_new_nw_config(self):
+        if self.__config["applianceControlPlaneIpAddress"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["networkForApplianceVM"].strip():
+            return False
+        if not self.__config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"].strip():
+            return False
+        if self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolStart"].strip():
+            return False
+        if self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolEnd"].strip():
+            return False
+        if self.__config["staticIpNetworkDetails"]["gatewayIPAddress"].strip():
+            return False
+        return True
+
+    def get_config_version(self):
+        if(self.check_if_old_nw_config()):
+            return Constant.OLD_CONFIG_VERSION
+        if(self.check_if_new_nw_config()):
+            return Constant.NEW_CONFIG_VERSION
+        raise InvalidInputError("Invalid Config provided")
```

Comment:
>InvalidInputError [](http://example.com/codeflow?start=14&length=17)

 It would be good to specify why the config is invalid.

Reply:
One naïve option is to say either (A,B,C) should be provided or (A,B,D,E) should be provided.  

There could be other ways too. 
Is the old config a super set of the new config?
If so, one option is to assume new config. if any of the older config data is present, assume old config. 

These are suggestions. We can do the first too. This just helps cx troubleshoot better on their own. hence adds value.

Reply:
Added the first solution as comment. 'Either provide values of only networkForApplianceVM, networkCIDRForApplianceVM in config file for network config or provide all values in config file.'

Reply:
Changed the comment to raise InvalidInputError("Either provide values of only networkForApplianceVM, networkCIDRForApplianceVM " + \
             "in config file or provide all values - networkForApplianceVM, networkCIDRForApplianceVM," + \
             " applianceControlPlaneIpAddress, k8sNodeIPPoolStart, k8sNodeIPPoolEnd, gatewayIPAddress for network config")